overlord/ifacestate/handlers.go

```diff
@@ -1386,9 +1386,17 @@ func (m *InterfaceManager) doAutoConnect(task *state.Task, _ *tomb.Tomb) error {
 	// if this is the case we can only proceed once the restart
 	// has happened or we may not have all the interfaces of the
 	// new core/base snap.
-	if err := snapstateFinishRestart(task, snapsup); err != nil {
+	needsFinishRestart := false
+	if err := task.Get("finish-restart", &needsFinishRestart); err != nil &&
```

Comment:
So in regards to the introduction of the following new task variables `finish-restart` and `prepare-restart` - what happens when an update, which needs to reboot, is scheduled before snapd is updated to this version?

Snapd updates first, then the other essential snaps update, but at this point the change does not have any task that sets either `prepare-restart` or `finish-restart`. What happens in this case? This is not entirely clear for me

Reply:
When the variables are not set (old snapd running when creating the change),  the behavior is the same as in the past, even with newer snapd running the tasks. What is an actual difference is that `discard-old-kernel-snap-setup` has moved but looking at the code there we are safe regardless of the order and the snapd version.

Said this, there is no kernel yet deployed using kernel-modules, so we are also safe if this gets into 2.67.

Reply:
sorry, I'm confused as if the finish-restart is not set needsFinishRestart is false, no? so we will not have the old behavior

Reply:
also wouldn't it make more sense to deal with the new field inside snapstate.FinishRestart itself? it receives the task anyway?